lib/chefdepartie/cookbook.rb

```diff
@@ -28,6 +33,7 @@ def self.upload_cookbooks(path, books)
 
       books = books.collect do |name|
         next if Cache.cache(File.join(path, name))
+        next if File.file?(File.join(path, name))
```

Comment:
I think there are conceivably edge cases, for special files (maybe for instance,  symlinks, where it might respond to File.file? with true, as well as File.directory? with true, if the target is a directory - unsure of this).